test/unit/ngsiv2/bidirectionalityHttp-test.js

```diff
@@ -0,0 +1,258 @@
+                    }
+
+                    latitudeFound.should.equal(true);
+                    longitudeFound.should.equal(true);
+
+                    done();
+                });
+            });
+        });
+    });
+
+    describe('When a bidirectional attribute is set and a new value arrives to a device with endpoint', function() {
+        beforeEach(function(done) {
+            var provisionOptions = {
+                url: 'http://localhost:' + config.iota.server.port + '/iot/devices',
+                method: 'POST',
+                json: utils.readExampleFile('./test/deviceProvisioning/provisionCommandBidirectionalWithUrl.json'),
+                headers: {
+                    'fiware-service': 'smartGondor',
+                    'fiware-servicepath': '/gardens'
+                }
+            };
+
+            nock.cleanAll();
+
+            contextBrokerMock = nock('http://192.168.1.1:1026')
+                .matchHeader('fiware-service', 'smartGondor')
+                .matchHeader('fiware-servicepath', '/gardens')
+                .post('/v2/subscriptions', function(body) {
+
+                    var expectedBody = utils.readExampleFile(
+                    './test/unit/ngsiv2/subscriptionRequests/bidirectionalSubscriptionRequest.json');
+                    // Note that expired field is not included in the json used by this mock as it is a dynamic
+                    // field. The following code performs such calculation and adds the field to the subscription
+                    // payload of the mock.
+                    if (!body.expires)
+                    {
+                        return false;
+                    }
+                    else if (moment(body.expires, 'YYYY-MM-DDTHH:mm:ss.SSSZ').isValid())
+                    {
+                        expectedBody.expires = moment().add(config.deviceRegistrationDuration);
+                        var expiresDiff = moment(expectedBody.expires).diff(body.expires, 'milliseconds');
+                        if (expiresDiff < 500) {
+                            delete expectedBody.expires;
+                            delete body.expires;
+
+                            return JSON.stringify(body) === JSON.stringify(expectedBody);
+                        }
+
+                        return false;
+                    }
+                    else {
+                        return false;
+                    }
+                })
+                .reply(201, null, {'Location': '/v2/subscriptions/51c0ac9ed714fb3b37d7d5a8'});
+
+            contextBrokerMock
+                .matchHeader('fiware-service', 'smartGondor')
+                .matchHeader('fiware-servicepath', '/gardens')
+                .post('/v1/updateContext', utils.readExampleFile(
+                    './test/contextRequests/createBidirectionalDevice.json'))
+                .reply(200, utils.readExampleFile(
+                    './test/contextResponses/createBidirectionalDeviceSuccess.json'));
```

Comment:
Like in other cases, I undesrtand that IOTA doesn't pay any attention to the updateContext response, so probably the readExampleFile() part could be replaced by just `{}` and use the usual comment above this lines:

```
// Note: /v1/updateContext response is not processed by IOTA so its content is irrelevant,
// as far as it is a 200 OK 
```

Not very important, but better to keep the mock pattern the same in all tests, for clarity.